sklearn/utils/tests/test_extmath.py

```diff
@@ -643,3 +644,154 @@ def test_stable_cumsum():
     assert_array_equal(stable_cumsum(A, axis=0), np.cumsum(A, axis=0))
     assert_array_equal(stable_cumsum(A, axis=1), np.cumsum(A, axis=1))
     assert_array_equal(stable_cumsum(A, axis=2), np.cumsum(A, axis=2))
+
+
+@pytest.mark.parametrize('dtype',
+                         (np.int32, np.int64, np.float32, np.float64))
+def test_randomized_pca_low_rank(dtype):
```

Comment:
Please consider integrating these 3 tests into the original ones from `randomize_svd`. I think that parametrizing the tests with either `randomized_svd` and `randomized_pca`, and centering X accordingly could do it pretty easily.

I'm concerned about so much duplicated code.